plugin/evm/vm_test.go

```diff
@@ -2110,3 +2125,158 @@ func TestCreateHandlers(t *testing.T) {
 		require.ErrorIs(t, elem.Error, rpc.ErrMissingBatchResponse)
 	}
 }
+
+// deployContract deploys the provided EVM bytecode using a prefunded test account
+// and returns the created contract address. It is reusable for any contract code.
+func deployContract(t *testing.T, vm *VM, gasPrice *big.Int, code []byte) common.Address {
+	callerAddr := vmtest.TestEthAddrs[0]
+	callerKey := vmtest.TestKeys[0]
+
+	nonce := vm.txPool.Nonce(callerAddr)
+	tx := types.NewTx(
+		&types.LegacyTx{
+			Nonce:    nonce,
```

Comment:
```suggestion
			Nonce:    nonce,
			To:       nil, // contract creation
```

Makes it explicit when reading the code.